src/librustc_mir/interpret/memory.rs

```diff
@@ -471,17 +519,15 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> Memory<'mir, 'tcx, M> {
             }
             _ => {
                 if let Ok(alloc) = self.get(id) {
-                    return Ok((Size::from_bytes(alloc.bytes.len() as u64), alloc.align));
+                    Ok((Size::from_bytes(alloc.bytes.len() as u64), alloc.align))
                 }
-                match liveness {
-                    InboundsCheck::MaybeDead => {
-                        // Must be a deallocated pointer
-                        self.dead_alloc_map.get(&id).cloned().ok_or_else(||
-                            ValidationFailure("allocation missing in dead_alloc_map".to_string())
-                                .into()
```

Comment:
I don't know why this previously raised an error instead of panicking when it did not find the allocation in the `dead_alloc_map`. That is in contradiction with the comment above the function (saying it will always return `Ok` for `liveness == MaybeDead`), there was no comment explaining this, and all tests still pass when turning this into an `unwrap`.

Reply:
We programmed "defensively" (error instead of ICE) in a few places in the beginnings of miri and it still shows in some places.

Reply:
I see. Yeah we also throw `Unsupported` in a few places like non-existent binops, where I think we can just ICE nowadays.